src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,227 @@
+                    else
+                        return true;
+                }
+            }
+            return true;
+        }
+
+
+        private async Task CreateGitHubIssueAsync(DateTime transactionDateTime , string repositoryUrl, string branch, 
+            string action, String errorMessage, String method, string whereToCreateIssue)
+        {
+            Logger.LogInformation($"Something failed in the repository : {repositoryUrl} ");
+
+            string fyiHandles = "@epananth";
+            string gitHubToken = null, azureDevOpsToken = null;
+            string label = "UpdateDependency";
+
+            using (Logger.BeginScope($"Opening GitHub issue for RepositoryBranchUpdateHistory."))
+            {
+                try
+                {
+                    if (!string.IsNullOrEmpty(whereToCreateIssue))
+                    {
+                        IGitHubTokenProvider gitHubTokenProvider = Context.GetService<IGitHubTokenProvider>();
+                        long installationId = await Context.GetInstallationId(whereToCreateIssue);
+                        gitHubToken = await gitHubTokenProvider.GetTokenForInstallationAsync(installationId);
+
+                        Logger.LogInformation($"GitHub token acquired for '{whereToCreateIssue}'!");
+                    }
+
+                    IssueManager issueManager = new IssueManager(gitHubToken, azureDevOpsToken);
+
+                    string title = $"Repository Url :'{repositoryUrl}' Error :'{errorMessage}'";
+                    string description = $"Something failed during dependency update" +
+                    $"{Environment.NewLine} {Environment.NewLine}" +
+                    "RepositoryUrl :" + $"[{repositoryUrl}](repositoryUrl)" +
+                    $"{Environment.NewLine} {Environment.NewLine}" +
+                    "Branch Name :" + $"{branch}" +
+                    $"{Environment.NewLine} {Environment.NewLine}" +
+                    "Error Message : " + $"{errorMessage}" +
+                    $"{Environment.NewLine} {Environment.NewLine}" +
+                    "Method : " + $"{method}" +
+                    $"{Environment.NewLine} {Environment.NewLine}" +
+                    "Action : " + $"{action}" +
+                    $"{Environment.NewLine} {Environment.NewLine}" +
+                    "Failure Category :" + $"{label}" + // Work on failure category will be done as a part of -> https://github.com/dotnet/arcade/issues/1196
+                    $"{Environment.NewLine} {Environment.NewLine}" +
+                    $"/FYI: {fyiHandles}";
+
+
+                    Logger.LogInformation($"GitHub token acquired for '{whereToCreateIssue}'!");
+                    //Check if the issue already exists in Github
+                    if (await SearchCreatedIssueAsync(transactionDateTime, errorMessage))
+                    {
+                        int issueId = await issueManager.CreateNewIssueAsync(whereToCreateIssue, title, description);
```

Comment:
The issue should probably also have some sort of label applied to them, so we can search them out and such more easily.  Though that would require not using the indirection of the "IssueManager"... which I assume is some Bar layer for dealing with issues and just using the GitHubClient.  We already force this to only work on GitHub (the searching, the "@" stuff), we the extra layer of indirection isn't buying us anything, and it's making is consider azdoTokens and being unable to label things.